src/dataset/mod.rs

```diff
@@ -222,6 +222,17 @@ pub trait AsTargets {
     }
```

Comment:
This trait should only have the method `as_single_target()` and should only be implemented for 1D datasets. Same with the mutable version of the trait.

Reply:
The `as_single_target` method shouldn't be fallible, so it won't have a default impl. Otherwise if we mark the wrong type with this trait we'd get panics. This trait should only be implemented on "single-target" types, such as `Array1`.